src/main.go

```diff
@@ -101,53 +111,74 @@ func Cmd() *cobra.Command {
 		"keyper set manager contract address",
 	)
 
+	cmd.PersistentFlags().IntVarP(
+		&Config.DelayFactor,
+		"delay-factor",
+		"",
+		10,
+		"Server cache delay factor",
+	)
+
 	return cmd
 }
 
 func Start() error {
+
+	wsClient, err := ethclient.Dial(Config.WebsocketURL)
+	if err != nil {
+		utils.Logger.Err(err).Msg("Failed to connect to websocket")
+	}
+
+	headers := make(chan *types.Header)
+
+	sub, err := wsClient.SubscribeNewHead(context.Background(), headers)
```

Comment:
This seems to be the wrong place to create the subscription. If I see this correctly, it's created here at main.go and then passed down to the server, the server does nothing with it but passing it down to the RPCService, and here it's only ever used in `HandleBlocks()`. 
IMO it could be created on the fly when starting HandleBlocks and passed to the routine. This way it would not be necessary to store it as a struct variable. 

Additionally, it should be mentioned that (AFAIK) there will always be only one consumer for each event on a channel. So if multiple go routines subscribe to a channel only one of them will actually consume it.

Reply:
Yes, although now I've removed this part since we will only rely on time events instead of listening to new blocks.